src/nvme/nbft.c

```diff
@@ -0,0 +1,947 @@
+#define SSNSFLAG_UNAVAILABLE_NAMESPACE_AVAIL	 0x0080
+#define SSNSFLAG_UNAVAILABLE_NAMESPACE_UNAVAIL	 0x0100
+#define SSNSFLAG_UNAVAILABLE_NAMESPACE_RESV	 0x0180
+
+#define SSNS_TCP_FLAG_VALID		0x01
+#define SSNS_TCP_FLAG_PDU_HEADER_DIGEST	0x02
+#define SSNS_TCP_FLAG_DATA_DIGEST	0x04
+
+/*
+ * SUBSYSTEM AND NAMESPACE EXTENDED INFORMATION DESCRIPTOR (Figure 19)
+ */
+struct __attribute__((__packed__)) raw_nbft_ssns_extended_info {
+	__u8 structure_id;
+	__u8 version;
+	__u16 ssns_index;
+	__u32 flags;
+	__u16 controller_id;
+	__u16 asqsz;
+	heap_obj dhcp_root_path_string;
+};
+
+#define SSNS_EXTINFO_FLAG_VALID		0x01
+#define SSNS_EXTINFO_FLAG_ADMIN_ASQSZ	0x02
+
+/*
+ * SECURITY DESCRIPTOR (Figure 21)
+ */
+struct __attribute__((__packed__)) raw_nbft_security {
+	__u8 structure_id;
+	__u8 index;
+	__u16 security_descriptor_flags;
+	__u8 secret_type;
+	__u8 reserved1;
+	heap_obj secure_channel_algorithm;
+	heap_obj authentication_protocols;
+	heap_obj cipher_suite;
+	heap_obj dh_groups;
+	heap_obj secure_hash_functions;
+	heap_obj secret_keypath;
+	__u8 reserved2[22];
+};
+
+#define SECFLAG_VALID(x)					0x0001 
+#define SECFLAG_IN_BAND_AUTHENTICATION_REQUIRED			0x0006
+#define SECFLAG_AUTHENTICATION_POLICY_LIST			0x0018
+#define SECFLAG_AUTHENTICATION_POLICY_LIST_NOT_SUP		 0x0000
+#define SECFLAG_AUTHENTICATION_POLICY_LIST_SUP			 0x0008
+#define SECFLAG_AUTHENTICATION_POLICY_LIST_REQ			 0x0010
+#define SECFLAG_AUTHENTICATION_POLICY_LIST_RSVD			 0x0018
+#define SECFLAG_SECURE_CHANNEL_NEGOTIATION			0x0060
+#define SECFLAG_SECURE_CHANNEL_NEGOTIATION_NOT_SUP		 0x0000
+#define SECFLAG_SECURE_CHANNEL_NEGOTIATION_SUP			 0x0020
+#define SECFLAG_SECURE_CHANNEL_NEGOTIATION_REQ			 0x0040
+#define SECFLAG_SECURE_CHANNEL_NEGOTIATION_RSVD			 0x0060
+#define SECFLAG_SECURITY_POLICY_LIST				0x0180
+#define SECFLAG_SECURITY_POLICY_LIST_NOT_PRES			 0x0000
+#define SECFLAG_SECURITY_POLICY_LIST_PRES			 0x0080
+#define SECFLAG_SECURITY_POLICY_LIST_PRES_ADMINSET		 0x0100
+#define SECFLAG_SECURITY_POLICY_LIST_RSVD			 0x0180
+#define SECFLAG_CIPHER_SUITES_RESTRICTED_BY_POLICY		0x0200
+#define SECFLAG_AUTH_DH_GROUPS_RESTRICTED_BY_POLICY_LIST	0x0400
+#define SECFLAG_SECURE_HASH_FUNCTIONS_POLICY_LIST		0x0800
+
+enum secret_type {
+	SECRET_TYPE_RESERVED,
+	SECRET_TYPE_REDFISH_HOST_INTERFACE_URI,
+};
+
+/*
+ * DISCOVERY DESCRIPTOR (Figure 24)
+ */
+struct __attribute__((__packed__)) raw_nbft_discovery {
+	__u8 structure_id;
+	__u8 flags;
+	__u8 index;
+	__u8 hfi_index;
+	__u8 security_index;
+	__u8 reserved1;
+	heap_obj discovery_ctrl_addr;
+	heap_obj discovery_controller_nqn;
+	__u8 reserved2[14];
+};
+
+#define DISCOVERYFLAG_VALID	0x01
+
+/*
+ *  End of NBFT ACPI table definitions
+ */
+#define MIN(a,b) (((a)<(b))?(a):(b))
+
+static __u8 csum(void *buffer, int length)
+{
+	int n;
+	__u8 sum = 0;
+
+	for (n = 0; n < length; n++) {
+		sum = (__u8)(sum + ((__u8 *)buffer)[n]);
+	}
+	return sum;
+}
+
+static void format_ip_addr(char *buf, size_t buflen, __u8 *addr) {
```

Comment:
For function the '{' goes on a new line.

Reply:
Fixed.